src/components/Dropdown.tsx

```diff
@@ -0,0 +1,282 @@
+/**
+ * A simple dropdown menu
+ * @author Alessandra De Lucas
+ * @author Gabe Abrams
+ */
+
+// Import React
+import React, { useReducer } from 'react';
+
+// Import FontAwesome
+// import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+// import { IconProp } from '@fortawesome/fontawesome-svg-core';
+
+import Variant from '../types/Variant';
+/*------------------------------------------------------------------------*/
+/* -------------------------------- Types ------------------------------- */
+/*------------------------------------------------------------------------*/
+
+// Props definition
+type Props = {
+  // currently selected dropdown item
+  selectedItem: DropdownItem,
+  // List of items in the dropdown menu
+  items: DropdownItem[],
+  // on click function for the dropdown button
+  onChange: (dropdownItem: DropdownItem) => void,
+  // If true, dropdown variant is displayed
+  variant?: Variant,
+  // If true, dropdown will only render content in the dropdown button
+  onlyRenderContentInButton?: boolean,
+  // Information about the dropdown button
+  // dropdownButton: {
+  //   // Text
+  //   text: string,
+  //   // Aria label for accessibility
+  //   ariaLabel: string,
+  //   // Unique ID
+  //   id: string,
+  //   // Element icon
+  //   icon?: IconProp,
+  // },
+  // If true, no arrow shows in dropdown
+  // noArrow?: boolean,
+  // Size of dropdown
+  // size?: DropdownSize,
+  // Direction of dropdown
+  // direction?: DropdownDirection,
+};
+
+// Details about a dropdown item
+// type DropdownItem = (
+//   | {
+//     // Header element
+//     type: DropdownItemType.Header,
+//     // Header text
+//     text: string,
+//   }
+//   | {
+//     // Divider element
+//     type: DropdownItemType.Divider,
+//   }
+//   | {
+//     // Item from dropdown menu
+//     type: DropdownItemType.Item,
+//     // Item text
+//     text: string,
+//     // Item aria label
+//     ariaLabel: string,
+//     // Unique item ID
+//     id: string,
+//     // Item icon
+//     icon?: IconProp,
+//     // Click function for dropdown item
+//     onClick: () => void,
+//   }
+// );
+
+type DropdownItem = {
+  // the text of the item
+  label: string,
+  // the value of the item (used for comparison)
+  value: any,
+  // optionally, a React node to render as the content of the item
+  // this will be rendered instead of the label, and the
+  // label will only be used for the aria-label
+  content?: React.ReactNode
+};
```

Comment:
Please restore the original version of this:

```
// Details about a dropdown item
type DropdownItem = (
  | {
    // Header element
    type: DropdownItemType.Header,
    // Header content
    content: React.ReactNode,
  }
  | {
    // Divider element
    type: DropdownItemType.Divider,
  }
  | {
    // Item from dropdown menu
    type: DropdownItemType.Item,
    // Item content
    content: React.ReactNode,
    // Item aria label
    ariaLabel: string,
    // Unique item ID
    id: string,
    // Item icon
    icon?: IconProp,
    // Click function for dropdown item
    onClick: () => void,
  }
);
```

Reply:
I did make some modifications here. Content is always `React.ReactNode`, so it'll be flexible enough for you. Then also, `ariaLabel` can be used for label building